FrontEnd/src/pages/Auth.tsx

```diff
@@ -0,0 +1,95 @@
+import { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { useAuthStore } from "../store/authStore";
+import LoginForm from "../components/LoginForm";
+import SignupForm from "../components/SignUpForm";
+
+export default function Login() {
+  const navigate = useNavigate();
+  const login = useAuthStore((state) => state.login);
+  const signup = useAuthStore((state) => state.signup);
+  const [isSignUp, setIsSignUp] = useState(false);
+
+  const handleSubmitLogin = (email: string, password: string) => {
+    login(email, password);
+    navigate("/profile");
+  };
+
+  const handleSubmitSignUp = (
+    name: string,
+    email: string,
+    password: string
+  ) => {
+    signup(email, password, name);
+    navigate("/profile");
+  };
+
+  return (
+    <div className="relative w-full h-screen overflow-hidden bg-transparent">
+      <div className="min-h-screen flex items-center justify-center text-gray-300 font-sans ">
+        <div className="relative w-full max-w-5xl overflow-hidden flex flex-col md:flex-row sm:px-6 md:px-0 md:bg-gray-800 md:bg-opacity-75 md:rounded-xl">
+          {/* Side Button */}
+          <div
+            className={`absolute inset-0 hidden md:flex justify-center items-center w-full md:w-1/2 transition-transform duration-700 ease-in-out ${
+              isSignUp ? "translate-x-0" : "translate-x-full"
+            }`}
+          >
+            {!isSignUp ? (
```

Comment:
Prepare this before the ` return ` and store the `JSX` in a variable. we need the ` return ` statement as short as we can. And let's avoid code duplication such as lines 38 & 52 and lines 50 & 54.